qubesadmin/vm/__init__.py

```diff
@@ -168,20 +173,21 @@ def get_power_state(self):
 
         '''
 
-        try:
-            vm_list_info = [line
-                for line in self.qubesd_call(
-                    self._method_dest, 'admin.vm.List', None, None
-                ).decode('ascii').split('\n')
-                if line.startswith(self._method_dest+' ')]
-        except qubesadmin.exc.QubesDaemonNoResponseError:
-            return 'NA'
-        assert len(vm_list_info) == 1
-        #  name class=... state=... other=...
-        # NOTE: when querying dom0, we get whole list
-        vm_state = vm_list_info[0].strip().partition('state=')[2].split(' ')[0]
-        return vm_state
-
+        if not self._state or force:
```

Comment:
Ouch, this will break all the long-running processes using `vm.is_running()` or such. One example is `qvm-shutdown --wait`. IMO caching of VM state should be opt-in, not opt-out.